src/main/java/dan200/computercraft/shared/util/InventoryUtil.java

```diff
@@ -169,7 +169,7 @@ private static ItemStack storeItems( ItemStack stack, IItemHandler inventory, in
             if( remainder == null ) break;
             remainder = inventory.insertItem( slot, remainder, false );
         }
-        return remainder;
+        return areItemsEqual( stack, remainder ) ? stack : remainder;
```

Comment:
for extra safety, line 166 could be "remainder = stack.copy()", incase the code modifies the itemstack in-place